File: python/shared/dtos/tier_dto.py
from . import BaseDTO, NameStr, NullableNameStr, NullableUrlStr


class TierDTO(BaseDTO):
    tier_id: int
    preset_id: int
    name: str
    icon_url: str | None

    model_config = {"from_attributes": True}


class AddTierDTO(BaseDTO):
    name: NameStr
    icon_url: NullableUrlStr


class UpdateTierDTO(BaseDTO):
    name: NullableNameStr = None
    icon_url: NullableUrlStr = None

File: web/src/dtos/tierDto.ts
import { z } from "zod";
import { nameSchema, nullableUrlSchema } from "@/utils/dto";

export const AddTierSchema = z.object({
  name: nameSchema,
  iconUrl: nullableUrlSchema,
});

export const UpdateTierSchema = z.object({
  name: nameSchema.optional(),
  iconUrl: nullableUrlSchema.optional(),
});

export type AddTierDTO = z.infer<typeof AddTierSchema>;
export type UpdateTierDTO = z.infer<typeof UpdateTierSchema>;

export interface TierDTO {
  tierId: number;
  presetId: number;
  name: string;
  iconUrl: string | null;
}
